### PR TITLE
[book] 1st to 4th ch changes

### PR DESCRIPTION
**Pull request by talitakz**
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | no |
| Applies to | all |
| Fixed tickets | non |

### CODE REVIEW

File: book/from_flat_php_to_symfony2.rst [hunk 1]
```diff
@@ -248,7 +256,7 @@ and reusable. To prove it, add a blog "show" page, which displays an individual
 blog post identified by an ``id`` query parameter.
 
 To begin, create a new function in the ``model.php`` file that retrieves
-an individual blog result based on a given id::
+an individual blog result based on a given ``id``::
```
Comment (stof) [inline book/from_flat_php_to_symfony2.rst]: I don't think this needs to be a code block here actually

File: book/from_flat_php_to_symfony2.rst [hunk 2]
```diff
@@ -399,13 +422,11 @@ routing, calling controllers, templates, etc.). More time will need to be
 spent to handle form submissions, input validation, logging and security.
 Why should you have to reinvent solutions to all these routine problems?
 
-.. _add-a-touch-of-symfony2:
```
Comment (stof) [inline book/from_flat_php_to_symfony2.rst]: this should not be removed, as it would break existing anchors (this is precisely why it was added)

File: book/from_flat_php_to_symfony2.rst [hunk 3]
```diff
@@ -683,7 +715,7 @@ Better Templates
 If you choose to use it, Symfony comes standard with a templating engine
 called `Twig`_ that makes templates faster to write and easier to read.
 It means that the sample application could contain even less code! Take,
-for example, the list template written in Twig:
+for example, lets rewrite ``layout.html.php`` template in Twig:
```
Comment (stof) [inline book/from_flat_php_to_symfony2.rst]: we never use ``lets`` in the doc AFAIK (and it should be ``let's`` anyway if you use it)

File: book/from_flat_php_to_symfony2.rst (hunk 2 above)
Comment (paxyknox) [inline book/from_flat_php_to_symfony2.rst]: I checked for the existing anchors and there are non, thou they might be on other branches. True.

File: book/from_flat_php_to_symfony2.rst [hunk 4]
```diff
@@ -705,7 +737,7 @@ for example, the list template written in Twig:
         </ul>
     {% endblock %}
 
-The corresponding ``layout.html.twig`` template is also easier to write:
+The corresponding ``layout.html.php`` template is also easier to write in Twig:
```
Comment (stof) [inline book/from_flat_php_to_symfony2.rst]: this is wrong. The template name ends with ``.twig``, not ``.php``

File: book/http_fundamentals.rst [hunk 5]
```diff
@@ -41,9 +41,6 @@ Symfony is built from the ground up around that reality. Whether you realize
 it or not, HTTP is something you use every day. With Symfony, you'll learn
 how to master it.
 
-.. index::
-   single: HTTP; Request-response paradigm
```
Comment (stof) [inline book/http_fundamentals.rst]: Why removing index entries ?

File: book/http_fundamentals.rst [hunk 6]
```diff
@@ -477,17 +499,14 @@ specific PHP method ``contactAction`` inside a class called ``MainController``::
...
-``<h1>Contact us!</h1>``. In the :doc:`controller chapter </book/controller>`,
+In this very simple example, the controller simply creates a ``Response`` object
+with the HTML ``<h1>Contact us!</h1>``. In the :doc:`controller chapter </book/controller>`,
 you'll learn how a controller can render templates, allowing your "presentation"
 code (i.e. anything that actually writes out HTML) to live in a separate
 template file. This frees up the controller to worry only about the hard
 stuff: interacting with the database, handling submitted data, or sending
 email messages.
 
-.. _symfony2-build-your-app-not-your-tools:
```
Comment (stof) [inline book/http_fundamentals.rst]: same issues here about BC for anchors

File: book/http_fundamentals.rst [hunk 7]
```diff
@@ -563,7 +575,7 @@ a PHP library that accomplishes two distinct tasks:
 
 The goal of the framework is to integrate many independent tools in order
 to provide a consistent experience for the developer. Even the framework
-itself is a Symfony bundle (i.e. a plugin) that can be configured or replaced
+itself is a Symfony :term:`bundle` (i.e. a plugin) that can be configured or replaced
```
Comment (stof) [inline book/http_fundamentals.rst]: I don't think the doc is using ``term`` anymore

File: book/from_flat_php_to_symfony2.rst (hunk 4 above)
Comment (paxyknox) [inline book/from_flat_php_to_symfony2.rst]: Actually if you look up closely , in these chapter you first write `layout.html.php` and `list.html.php` and then, where i made a change, you rewrite these to templates with Twig.

File: book/page_creation.rst [hunk 8]
```diff
@@ -1,8 +1,5 @@
 .. index::
-   single: Page creation
-
-.. _creating-pages-in-symfony2:
-.. _creating-pages-in-symfony:
```
Comment (stof) [inline book/page_creation.rst]: these anchors must be preserved.

File: book/from_flat_php_to_symfony2.rst (hunk 2 above)
Comment (xabbuh) [inline book/from_flat_php_to_symfony2.rst]: Anchors can be used by other sites to link to particular sections of the docs. That's why we keep them.

File: book/page_creation.rst (hunk 8 above)
Comment (paxyknox) [inline book/page_creation.rst]: Noted!

File: book/from_flat_php_to_symfony2.rst (hunk 2 above)
Comment (stof) [inline book/from_flat_php_to_symfony2.rst]: Have you checked the whole Internet for links pointing to this page and using an anchor ?

File: book/http_fundamentals.rst (hunk 7 above)
Comment (paxyknox) [inline book/http_fundamentals.rst]: Do you mean new branches?

File: book/from_flat_php_to_symfony2.rst (hunk 2 above)
Comment (paxyknox) [inline book/from_flat_php_to_symfony2.rst]: I didn't know that. Noted.

File: book/http_fundamentals.rst (hunk 7 above)
Comment (stof) [inline book/http_fundamentals.rst]: Ah no. Looks like https://github.com/symfony/symfony-docs/pull/5464 has not been merged

File: book/http_fundamentals.rst (hunk 5 above)
Comment (paxyknox) [inline book/http_fundamentals.rst]: Question? Is this used anywhere? They are placed so confusingly. They are just above some  (sub)titles.

Sorry for asking.. When I correct the mistakes and undo some thing I just normally push o do I have to make another pull request? My first time contributing, may I be forgiven :)

File: book/from_flat_php_to_symfony2.rst [hunk 9]
```diff
@@ -5,10 +8,11 @@ Symfony versus Flat PHP
 
 **Why is Symfony better than just opening up a file and writing flat PHP?**
 
-If you've never used a PHP framework, aren't familiar with the MVC philosophy,
-or just wonder what all the *hype* is around Symfony, this chapter is for
-you. Instead of *telling* you that Symfony allows you to develop faster and
-better software than with flat PHP, you'll see for yourself.
+If you've never used a PHP framework, aren't familiar with the
+model-view-controller `MVC`_ philosophy, or just wonder what all the *hype*
```
Comment (WouterJ) [inline book/from_flat_php_to_symfony2.rst]: I would change this to: "Model-View-Controller (MVC) philosophy" also, we don't use a serial comma (the one before "or") in the docs. Although that isn't your fault, it would be nice to include it in this PR

File: book/from_flat_php_to_symfony2.rst [hunk 10]
```diff
@@ -295,8 +305,8 @@ this page introduces even more lingering problems that a framework can solve
 for you. For example, a missing or invalid ``id`` query parameter will cause
 the page to crash. It would be better if this caused a 404 page to be rendered,
 but this can't really be done easily yet. Worse, had you forgotten to clean
-the ``id`` parameter via the ``intval()`` function, your
-entire database would be at risk for an SQL injection attack.
+the ``id`` parameter via the PHP ``intval()`` function, your
```
Comment (WouterJ) [inline book/from_flat_php_to_symfony2.rst]: I think the readflow is a bit better if we put "PHP" after the function name: "`intval()` PHP function"

File: book/from_flat_php_to_symfony2.rst [hunk 11]
```diff
@@ -362,8 +373,12 @@ on the requested URI::
     }
 
 For organization, both controllers (formerly ``index.php`` and ``show.php``)
-are now PHP functions and each has been moved into a separate file, ``controllers.php``::
+are now PHP functions and each has been moved into a separate file named
+``controllers.php``. The job of each PHP function, now called a
+:term:`controller`, is to use information from the ``Request`` object to create
+and return a ``Response`` object.::
```
Comment (WouterJ) [inline book/from_flat_php_to_symfony2.rst]: either end with double colon (which will render as a single one) or end with a final stop (`.`) and use `.. code-block:: php` on a new line

File: book/from_flat_php_to_symfony2.rst (hunk 3 above)
Comment (WouterJ) [inline book/from_flat_php_to_symfony2.rst]: indeed, we should avoid first person usage in all docs (apart from best practices and quick tour)

File: book/from_flat_php_to_symfony2.rst (hunk 4 above)
Comment (WouterJ) [inline book/from_flat_php_to_symfony2.rst]: To be honest, I don't fully get your changes here. Line 718 now says "lets rewrite ``layout.html.php`` template in Twig:" and directly after it, you have this line also mentioning "The corresponding ``layout.html.php`` template is also easier to write in Twig:"

File: book/http_fundamentals.rst (hunk 5 above)
Comment (WouterJ) [inline book/http_fundamentals.rst]: @paxyknox some people are reviewing lots of PRs each days, so they may sound a bit to the point. We are very happy with your PR and having 10 wrong changes in a PR containing 1000 changes is still not bad :wink: 

In this case, indexes are used on the index of the documentation to redirect people to sections of the docs: http://symfony.com/doc/current/genindex.html While I can't see much references to it, I think we should keep the indexes as they were for now.

Comment (paxyknox) [inline book/http_fundamentals.rst]: Well page is something I never came across :) Thank you! Clarified a few things, yes.

File: book/from_flat_php_to_symfony2.rst (hunk 4 above)
Comment (paxyknox) [inline book/from_flat_php_to_symfony2.rst]: Looked again. On the 718th line it should be `list.html.php` and then on the 740th line it should be `layout.html.php` but without "corresponding" which adds a bit of confusion.
The chapter flow is:
1) create list and layout .php file
2) rewrite them in *.html.php
3) and finally rewrite them in *.html.twig

**Comment by paxyknox** [comment]
I made requested changes.
If there are no other remarks this could probably be merged.

File: book/from_flat_php_to_symfony2.rst [hunk 12]
```diff
@@ -98,11 +102,12 @@ the code that prepares the HTML "presentation"::
     require 'templates/list.php';
 
 
-The HTML code is now stored in a separate file (``templates/list.php``), which
+The HTML code is now stored in a separate file ``templates/list.php``, which
```
Comment (WouterJ) [inline book/from_flat_php_to_symfony2.rst]: Changing the order to "in a separate `templates/list.php` file" seems to be better (although I'm not a native).

File: book/from_flat_php_to_symfony2.rst [hunk 13]
```diff
@@ -557,10 +581,15 @@ them for you. Here's the same sample application, now built in Symfony::
         }
     }
 
+First we have a "controller class" which is a convenient way to group several
+"controllers" together. So methods inside a controller class are controllers
+also called *actions*. They hold code which creates and returns the appropriate
+``Response`` object.
```
Comment (WouterJ) [inline book/from_flat_php_to_symfony2.rst]: I'm not sure we should talk about controllers and actions and their differences at this point. It's just a very very quick introduction article into Symfony. Such detailed things should be discussed in the main book chapters (e.g. "Controllers") imo

File: book/from_flat_php_to_symfony2.rst [hunk 14]
```diff
@@ -635,55 +665,27 @@ even need to create it!)::
...
-The front controller's only job is to initialize Symfony's engine (``Kernel``)
-and pass it a ``Request`` object to handle. Symfony's core then uses the
-routing map to determine which controller to call. Just like before, the
-controller method is responsible for returning the final ``Response`` object.
-There's really not much else to it.
+Front controller's only job is to initialize Symfony's engine (called the
+:term:`Kernel`) and pass it a ``Request`` object to handle. The Symfony core
+asks the router to inspect the request. The router matches the incoming URL
+to a specific route and returns information about the route, including the
+controller that should be executed. The correct controller from the matched
```
Comment (WouterJ) [inline book/from_flat_php_to_symfony2.rst]: I consider this also too specific and too focussed on the implementation for a getting started docs.

File: book/http_fundamentals.rst (hunk 5 above)
Comment (WouterJ) [inline book/http_fundamentals.rst]: You seem to have forgotten to revert this change

File: book/from_flat_php_to_symfony2.rst (hunk 13 above)
Comment (paxyknox) [inline book/from_flat_php_to_symfony2.rst]: I added this based on my personal experience... When i was introduced to Symfony and start reading the documentation i had trouble understanding this chapter and the controller concept - diff between flat PHP and Symfony controllers - because of this missing information that didn't appeared until in some later chapter.. World controller is used thought the chapter and it does not always refer to the same thing. So i think that to a newcomer it's quite unclear if it doesn't have any background from before and essential to understand what it's actually meant.
Do you think i should still through this out?

File: book/http_fundamentals.rst (hunk 5 above)
Comment (paxyknox) [inline book/http_fundamentals.rst]: True.

File: book/from_flat_php_to_symfony2.rst (hunk 14 above)
Comment (paxyknox) [inline book/from_flat_php_to_symfony2.rst]: Do u want me to just revert or should I also remove the original text with (or without) the picture?

File: book/from_flat_php_to_symfony2.rst (hunk 12 above)
Comment (paxyknox) [inline book/from_flat_php_to_symfony2.rst]: I agree.

**Comment by paxyknox** [comment]
I would appreciate a final response so this could be finally merged. Thank you!

**Comment by weaverryan** [comment]
Hey @paxyknox!

Ok, I've just merged this in! Actually, I created a new PR - #6743 - so I could rebase, make changes their and see the diff. 

The reviews you did and PR's you made are great... but also really tough for us (hence the long delay). Merging *just* this PR took me almost 2 hours. The problem is their size: your PR's are *huge*, and contain a lot of good changes, but also a lot of changes that I need to remove (e.g. the `::` that you added in many places is actually incorrect).

So unfortunately, I will be closing all of your PR's - we simply can't review them in a timely fashion. This is a shame because there's good stuff in there.

In the future, to help us out, here's what we need:

A) Smaller PR's, both in general size and in how many changes show up. For example, if you change the indentation on a section (or change something from a sidebar to *not* be a sidebar), that change looks huge in the PR. If you *only* made a PR with this change and said "I just changed this to *not* be a sidebar, but didn't update the text" - then that's an easy change :).

B) Organize PR's by *what* your modifying. For example, if you're rewording some things to improve their wording, *only* do that - don't also fix other things. Or, if you're adding more links or `:class` links to be helpful, *only* add those. When things are mixed, simple, nice improvements get blocked while we discuss and try to modify the bigger stuff. Or, conversely, a nice re-wording might be blocked by other incorrect changes (like adding the `::` in places where it shouldn't be).

Cheers!
